lib/puppet/provider/service/init.rb

```diff
@@ -152,6 +152,15 @@ def restartcmd
     (@resource[:hasrestart] == :true) && [initscript, :restart]
   end
 
+  def texecute(type, command, fof = true, squelch = false, combine = true)
+    if type =~ /^start/
+      if Facter.value(:operatingsystem) == "Solaris"
+        command =  ["/usr/bin/ctrun -l none", command].flatten.join(" ")
```

Comment:
Are we sure this `flatten.join()` will play nice with all the possible inputs here?